### ISSUE TITLE
How to use setCategoricalSlotNames in LightGBMClassifier

### ISSUE THREAD

**Issue by mchl0203** [Azure/mmlspark#698]
**version**
I compiled the latest version of mmlspark on 9-26, and use lightgbmlib == 2.2.400.

**operation**
I think the following attempt shows that setCategoricalSlotNames can't work. Because I can't find enumeration value in feature_infos of model file.
I have a dataframe with 4 columns A, B, C, D.
1. setFeaturesCol( VectorAssembler(A,B,C,D)) and setCategoricalSlotNames(Array(D))
2. setFeaturesCol( VectorAssembler(A,B,C,D)) and setCategoricalSlotNames(Array("Column_3"))
3. setFeaturesCol( VectorAssembler(A,B,C)) and setCategoricalSlotNames(Array(D))

The setCategoricalSlotIndexes is ok, when I  setFeaturesCol( VectorAssembler(A,B,C,D)) and setCategoricalSlotIndexes(Array(3))

Looking forward to your reply, thank you!

**Comment by mchl0203** [comment]
Sorry, The reason setCategoricalSlotNames can't work is that I write a transformer to replace VectorAssembler(The code has been written for too long and I has forgotten). If I use VectorAssembler, setCategoricalSlotNames is ok. I don't know why LightGBM need VectorAssembler, I am reading at the mmlspark related code. So If I know the reason, I will close this issue. If anyone knows why, please tell me. Thank you！

**Comment by imatiach-msft** [comment]
hi @mchl0203 , LightGBM does not require vector assembler or any other preprocessing steps, it just needs a vector of values for the features column (similar to all other Spark ML learners) and a column of double values for the labels.  Hope that clears up any confusion.  If you are still having trouble, please post a repro with the code and dataset and I can take a look.  There was an issue with categorical features which was recently fixed, but it sounds like you have the latest compiled code.

**Comment by mchl0203** [comment]
hi @imatiach-msft Thank you for your reply, I know why setCategoricalSlotNames can't work when I used. The vector column after VectorAssembler contains metadata(), but I did not add matedata to this transformer that I implemented myself. So feature_names in the model file becomes the following image, setCategoricalSlotNames can't take effect. 

![image](https://user-images.githubusercontent.com/8417221/65944899-4d3a3180-e465-11e9-8231-d2cd1e3210b7.png)
When I add the metadata(Operate as in the transform function in the VectorAssembler.scala file ), setCategoricalSlotNames is ok~
